File: backend/app/schemas/sustainability.py
"""Pydantic schemas for plant sustainability / CO₂ savings responses."""

from pydantic import BaseModel


class SustainabilitySummary(BaseModel):
    """Lifetime CO₂ savings estimate for a single solar plant."""

    plant_id: int
    total_kwh_generated: float
    co2_saved_kg: float
    co2_saved_tonnes: float
    equivalence_statement: str

File: frontend/src/api/sustainability.ts
/**
 * Sustainability API calls — lifetime CO₂ savings for a single plant.
 */

import apiClient from './client'
import type { SustainabilitySummary } from '@/types'

export async function getPlantSustainability(
  plantId: number,
): Promise<SustainabilitySummary> {
  const { data } = await apiClient.get<SustainabilitySummary>(
    `/plants/${plantId}/sustainability`,
  )
  return data
}
